bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/discovery/VenstarThermostatDiscoveryService.java

```diff
@@ -0,0 +1,249 @@
+        }
+
+        return null;
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and scans them for the
+     * search keywords. The search is not case sensitive.
+     *
+     * @param socket
+     *            The socket where the answers arrive.
+     * @param keywords
+     *            The keywords to be searched for.
+     * @return
+     * @throws IOException
+     */
+    private void scanResposesForKeywords(MulticastSocket socket, String... keywords) throws IOException {
+        // In the worst case a SocketTimeoutException raises
+        do {
+            byte[] rxbuf = new byte[8192];
+            DatagramPacket packet = new DatagramPacket(rxbuf, rxbuf.length);
+            try {
+                socket.receive(packet);
+            } catch (IOException e) {
+                logger.trace("Got exception while trying to receive UPnP packets: {}", e.getMessage());
+                return;
+            }
+            String response = new String(packet.getData());
+            if (response.contains(SSDP_MATCH)) {
+                logger.trace("Match: {} ", response);
+                parseResponse(response);
+            }
+        } while (true);
+    }
+
+    protected void parseResponse(String response) {
+        DiscoveryResult result;
+
+        String name = null;
+        String url = null;
+        String uuid = null;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.trace("key: {} value: {}.", key, value);
+            switch (key) {
+                case "location":
+                    url = value;
+                    break;
+                case "usn":
+                    Matcher m = USN_PATTERN.matcher(value);
+                    if (m.find()) {
+                        uuid = m.group(2);
+                        name = m.group(3);
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+        scanner.close();
+
+        logger.trace("Found thermostat, name: {} uuid: {} url: {}", name, uuid, url);
+
+        if (name == null || uuid == null || url == null) {
+            logger.trace("Bad Format from thermostat");
+            return;
+        }
+
+        uuid = uuid.replace(":", "").toLowerCase();
+
+        ThingUID thingUid = new ThingUID(VenstarThermostatBindingConstants.THING_TYPE_COLOR_TOUCH, uuid);
+
+        logger.trace("Got discovered device.");
+
+        String label = String.format("Venstar Thermostat (%s)", name);
+        result = DiscoveryResultBuilder.create(thingUid).withLabel(label).withRepresentationProperty(uuid)
```

Comment:
```suggestion
        result = DiscoveryResultBuilder.create(thingUid).withLabel(label).withRepresentationProperty(VenstarThermostatBindingConstants.PROPERTY_UUID)
```

I did notice this one recently as my 3.x system is complaining about it this property if its missing/wrong.